stdlib/src/collections/dict.mojo

```diff
@@ -958,7 +967,7 @@ struct Dict[K: KeyElement, V: CollectionElement](
             while not self._entries.__get_ref(right)[]:
                 right += 1
                 debug_assert(right < self._reserved(), "Invalid dict state")
-            var entry = self._entries.__get_ref(right)
+            var entry = Reference(self._entries[right])
```

Comment:
This change isn't related, and it's also going to make a copy the way `List.__getitem__` is currently implemented. I think `List` will be fixed soon :tm: but let's not touch this for now.

Reply:
Just wanted to follow-up here: @bethebunny was totally right that up until _very recently_ this change would have done a copy, but since `List.__getitem__` was changed to return a `ref` just a couple days ago, this change to use `Reference(..)` is the right one 🙂 

So carry on @rd4com, your line is correct: `Reference(self._entries[right])`.